Interpolation tutorial

Balaje opened the pull request:
Hi @ericneiva @santiagobadia,

I am submitting this PR on a tutorial for the interpolation feature. It's an extension of [blog-script.jl](https://gist.github.com/Balaje/fa9769bff8fc29945e1220aa29e93afa). I still have to add examples for different reffes, dimension, vector problems etc.

ericneiva commented on deps/build.jl:
```diff
@@ -18,7 +18,8 @@ files = [
   "Low-level API Poisson equation"=>"poisson_dev_fe.jl",
-  "On using DrWatson.jl"=>"validation_DrWatson.jl"]
+  "On using DrWatson.jl"=>"validation_DrWatson.jl",
+  "Interpolation"=>"interpolation_fe.jl"]
```
I would title it "Interpolation of `CellFields`" to be more specific.

ericneiva commented on src/interpolation_fe.jl:
```diff
@@ -0,0 +1,176 @@
+
+# Let us define the infinite dimensional function
+
+f₁(x) = x[1] + x[2]
```
Maybe rename "f_1" and "f_2" below to simply "f" to avoid association with the FE space numbering.

ericneiva commented on src/interpolation_fe.jl:
```diff
@@ -0,0 +1,176 @@
+
+# Like earlier we can check our results
+
+gₕ(pt), f₂(pt)
```
I assume you removed the tests because they are failing... I would change the problem such that the tests pass, instead of removing the tests.

ericneiva replied:
You can try just moving between linear spaces (i.e. linear V_2), you have already used `order = 2` in Lagrangian.

ericneiva commented on src/interpolation_fe.jl:
````diff
@@ -0,0 +1,176 @@
+# using the composite type `Interpolable` to tell `Gridap` that the
+# argument can be interpolated between triangulations. This can be
+# done using any of the methods
+# ```julia
````
I think this risks not ageing well... because, if for some reason we change the name of `interpolate` or `interpolate_dirichlet`, it will be hard to find out that this comment is outdated. I would remove this comment block and below, when you first call `interpolate_everywhere`, point out that you can also do `interpolate` or `interpolate_dirichlet` and execute both in a cell. In this way, if these procedures change, this tutorial will also fail.

ericneiva reviewed (changes requested):
Hi, @Balaje, good job preparing the tutorial and adding more explanation. I think you don't need to add more tests. In addition, to the changes I request you in the code, I also propose you the following changes:

* [ ] Remove `Manifest.toml` file from project and update Gridap compat entry in the `Project.toml` to 0.16.4.
* [ ] Rename `assets/interpolation/target.png` to `assets/interpolation_fe/source_and_target.png`. Note that I would also rename the folder.
* [ ] Put the `.DS_Store` extension in the `.gitignore` file.
* [ ] Maybe add an acknowledgements section at the end of the tutorial stating something like (you can change it) "Gridap contributors acknowledge support received from Google, Inc. through the GSoC 2021 project $TITLE_OF_PROJECT$".

Balaje commented:
Hi @ericneiva,

Sorry for the delay in updating this. I had to do some work writing my thesis 😞.

> I would still change the second mesh (or introduce a third one) or any other parameter such that the RT test passes.
 
I decided to remove the `sinusoidal` map, and leave it as a uniform partition with a different mesh size. Now, I think it should not fail that often. 
https://github.com/gridap/Tutorials/blob/7f79a062beb22407c1d4925db7e12b36dc86dfdb/src/interpolation_fe.jl#L87

Also, I changed the title:

https://github.com/gridap/Tutorials/blob/7f79a062beb22407c1d4925db7e12b36dc86dfdb/deps/build.jl#L22

I also added some details on interpolating `VectorValued` and `MultiFieldFEFunction` after the RT part.

https://github.com/gridap/Tutorials/blob/7f79a062beb22407c1d4925db7e12b36dc86dfdb/src/interpolation_fe.jl#L183
https://github.com/gridap/Tutorials/blob/7f79a062beb22407c1d4925db7e12b36dc86dfdb/src/interpolation_fe.jl#L215

I think the tutorial has everything I mentioned at the start (except for maybe higher dimension). Please let me know if I need to add/change anything else. 

Thanks! 🙂

ericneiva commented:
Hi, @Balaje, 

> Sorry for the delay in updating this. I had to do some work writing my thesis.

It's alright! Take your time :slightly_smiling_face: 

> I decided to remove the sinusoidal map

Have you updated the `../assets/interpolation_fe/source_and_target.png` picture accordingly?

After checking this :point_up: , you can mention Santi below, to ask him to review and merge the PR. I do not have permissions to do it.

Balaje commented:
>Have you updated the ../assets/interpolation_fe/source_and_target.png picture accordingly?

@ericneiva Yes, I have. Here is the file [`../assets/interpolation_fe/source_and_target.png`](https://github.com/Balaje/Tutorials/blob/interpolation_tutorial/assets/interpolation_fe/source_and_target.png). 

Hi @santiagobadia,

Can you please review this PR and merge if everything is working?

Thanks!

Balaje commented:
Hi @ericneiva @santiagobadia,

Yes. I have updated the image - [`/assets/interpolation_fe/source_and_target.png`](https://github.com/Balaje/Tutorials/blob/interpolation_tutorial/assets/interpolation_fe/source_and_target.png). 

@santiagobadia, can you please review this PR and merge if everything is fine?

Thanks!

fverdugo commented on Manifest.toml:
```diff
@@ -0,0 +1,1121 @@
+# This file is machine-generated - editing it directly is not advised
+
```
Hi @Balaje!

please do not commit the Manifest.toml
